### ISSUE TITLE
Website title should be linked

### ISSUE THREAD

**Issue by cardil** [squidfunk/mkdocs-material#2815]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

The main title on the website isn't linked. It should be linked to the home page, the same as the logo does.

Such link is expected by many users, and without it, the page title is useless. It has no function.

We stumbled upon this issues in knative/docs#3945

### Expected behaviour

Clicking the page title should link to homepage.

### Actual behaviour

The page title is a useless, unclickable text.

### Steps to reproduce

1. Go to any page
2. Try to click on title

### Package versions

- Python: `3.9.5`
- MkDocs: `1.2.1`
- Material: `7.1.9`

### Configuration

```yaml
site_name: Example
```

### System information

- Operating system: Fedora 34
- Browser: Firefox 89.0.2

**Comment by cardil** [comment]
Closed issue #2044 that looks like having some traction.

**Comment by squidfunk** [comment]
Please see https://github.com/squidfunk/mkdocs-material/issues/2044#issuecomment-724072138 for an explanation why, IMHO, this is not a good idea.

**Comment by cardil** [comment]
I did read that comment, but I think it's not completely valid. 

> _As the title bar is very narrow on iOS Safari, the user might miss the title bar and tap onto the header link, taking him to the home page._

Why do you think that adding a link to the title could suppress user's ability to hit other links? They have separate hit boxes, and those do not influence each other.

> _Furthermore, there would be two links, the icon in the sidebar and the title bar both taking the user to the homepage._

What's wrong with having two or even more links leading to one place? That's normal to have more than one link in many cases.

> _For that reason, your proposed solution doesn't cater well to all screen sizes._

I don't feel like I'm buying your arguments. Maybe I haven't understood your point. Please, could you rephrase one again your opinion, so it will be more clear?

**Comment by squidfunk** [comment]
> Why do you think that adding a link to the title could suppress user's ability to hit other links? They have separate hit boxes, and those do not influence each other.

With title bar I really meant top bar, the collapsed browser bar which hosts the carrier info, clock and website name. This top bar can be very narrow, at least on iOS, and a link target below it might be too hard to miss when tapping with your thumb on the top bar, holding the phone in your hand.

> What's wrong with having two or even more links leading to one place? That's normal to have more than one link in many cases.

Well yes, maybe it's just a matter of taste.

> I don't feel like I'm buying your arguments. Maybe I haven't understood your point. Please, could you rephrase one again your opinion, so it will be more clear?

Some other things to consider:

1. The icon link which symbolizes "home" can be set to a different link via [`extra.homepage`](https://squidfunk.github.io/mkdocs-material/setup/changing-the-logo-and-icons/?h=extra+homep#logo). This would also be the URL we would be using for the link in the document title you're proposing, which might not what some users might expect.
2. When scrolling, the site title swaps to the page title, which would not have this link (as per your proposal and PR). To me, this doesn't feel consistent, as the page title now isn't clickable anymore, but also stands side-by-side with the logo.
3. I feel that it's too late to make this change because it would mean that some users will want the old behavior back, as this has been with some changes. Changes that impact the top bar can be quite controversial, as on mobile, it's the entrypoint to navigation. I don't want to maintain two solutions for this little detail and add another configuration option.
4. It's easy to add via [theme extension](https://squidfunk.github.io/mkdocs-material/customization/)

Thus, I kindly ask you to respect my decision, but IMHO this isn't a good idea to consider to be adding to `master`. If you want this behavior, I'd recommend using theme extension, which is very easy to set up.

**Comment by cardil** [comment]
Thanks for your explanation. I see it differently, but I'll not push it any further, here.

What I'd like to do instead is to lower the boundary to change it. Currently, we've been forced to copy the entire [header.html partial file](https://github.com/squidfunk/mkdocs-material/blob/master/material/partials/header.html), just to replace `span` with `a`. That's a big burden on downstream projects.

Do you think it's possible to split the header partial file, in a way that it makes more easy to tweak the logo area?

**Comment by squidfunk** [comment]
> Do you think it's possible to split the header partial file, in a way that it makes more easy to tweak the logo area?

The logo as such is already located in `logo.html`, as it is reused in the navigation drawer.

Splitting stuff up further IMHO doesn't make much sense, as it will slow down the compilation of large documentation projects (some have 1.000+ pages) due to the [cost of further indirections](https://github.com/squidfunk/mkdocs-material/pull/2213) and I don't see many use cases for putting the header title into a single partial except the one you raised in this issue.
